src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -377,3 +379,363 @@ class Config:
+            else:
+                data['type'] = 'Disk'
+        elif extent_type == 'FILE':
+            data['type'] = 'File'
+
+        return data
+
+    @private
+    async def extend(self, data):
+        extent_type = data['type']
+
+        if extent_type != 'File':
+            # ZVOL and HAST are type DISK
+            data['type'] = 'DISK'
+        else:
+            extent_size = data['filesize']
+
+            # Legacy Compat for having 2[KB, MB, GB, etc] in database
+            if not extent_size.isdigit():
+                suffixes = {
+                    'PB': 1125899906842624,
+                    'TB': 1099511627776,
+                    'GB': 1073741824,
+                    'MB': 1048576,
+                    'KB': 1024,
+                    'B': 1
+                }
+                for x in suffixes.keys():
+                    if extent_size.upper().endswith(x):
+                        extent_size = extent_size.upper().strip(x)
+                        extent_size = int(extent_size) * suffixes[x]
+
+                        data['filesize'] = extent_size
+
+        data['type'] = extent_type.upper()
+
+        return data
+
+    @private
+    async def clean(self, data, schema_name, verrors):
+        await self.clean_name(data, schema_name, verrors)
+        await self.clean_type_and_path(data, schema_name, verrors)
+        await self.clean_size(data, schema_name, verrors)
+
+    @private
+    async def clean_name(self, data, schema_name, verrors):
+        name = data['name']
+        serial = data['serial']
+        name_filters = [('name', '=', name)]
+
+        if '"' in name:
+            verrors.add(f'{schema_name}.name', 'Double quotes are not allowed')
+
+        if '"' in serial:
+            verrors.add(f'{schema_name}.serial',
+                        'Double quotes are not allowed')
+
+        name_result = await self.middleware.call(
+            'datastore.query', self._config.datastore,
+            name_filters,
+            {'prefix': self._config.datastore_prefix})
+
+        if name_result:
+            verrors.add(f'{schema_name}.name', 'Extent name must be unique')
+
+    @private
+    async def clean_type_and_path(self, data, schema_name, verrors):
+        extent_type = data['type']
+        disk = data['disk']
+        path = data['path']
+        pools = await self.middleware.call('pool.query')
+        valid = False
+
+        if extent_type is None:
+            return data
+
+        if extent_type == 'Disk':
+            if not disk:
+                verrors.add(f'{schema_name}.type', 'This field is required')
+            if disk.startswith('zvol') and not os.path.exists(f'/dev/{disk}'):
+                verrors.add(f'{schema_name}.disk',
+                            f'ZVOL {disk} does not exist')
+        elif extent_type == 'File':
+            if not path:
+                verrors.add(f'{schema_name}.path', 'This field is required')
+
+            if 'iocage' in path:
```

Comment:
That does not seem to be a good enough check, regex `/iocage/?` would probably make more sense